### PR TITLE
Expressions+Testdata: Create json schema for query types

### PR DESCRIPTION
**Pull request by ryantxu**
(extracted from https://github.com/grafana/grafana/pull/82391)

This builds schemas for expression queries and saves them in a local file.  The schemas will be attached to the query service apiserver in another PR.

It is still unclear to me if including these files is useful or not:
image
The schemas target two different usages (in panels and in query) both require more properties to be useful as standalone schemas.

### CODE REVIEW

File: pkg/expr/query_test.go [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+					}),
+				},
+			},
+		},
+		schemabuilder.QueryTypeInfo{
+			Discriminators: data.NewDiscriminators("queryType", QueryTypeClassic),
+			GoType:         reflect.TypeOf(&ClassicQuery{}),
+			Examples:       []data.QueryExample{
+				// {
+				// 	Name: "do classic query (TODO)",
```
[the remark this answers is not in the document]
Comment (kylebrandt) [reply pkg/expr/query_test.go:99]: Are the `ConditionsCmd.Conditions` in https://github.com/grafana/grafana/blob/main/pkg/expr/classic/classic_test.go what you are looking for perhaps?

Or 

image

?

File: pkg/expr/query_test.go [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+						  }
+						]
+					  }`),
+				},
+			},
+		},
+	)
+
+	require.NoError(t, err)
+	_ = builder.UpdateQueryDefinition(t, "./")
```
Comment (marefr) [inline pkg/expr/query_test.go:167]: Nit. We should use proper go codegen rather thsn tests.

File: pkg/tsdb/grafana-testdata-datasource/kinds/query_test.go [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+							},
+						},
+					),
+				},
+			},
+		},
+	)
+
+	require.NoError(t, err)
+	builder.UpdateQueryDefinition(t, "./")
```
Comment (marefr) [inline pkg/tsdb/grafana-testdata-datasource/kinds/query_test.go:61]: Nit. We should use proper go codegen rather thsn tests.

File: pkg/tsdb/grafana-testdata-datasource/kinds/model.go
Comment (marefr) [inline pkg/tsdb/grafana-testdata-datasource/kinds/model.go:1]: Nit. Would be good if these types was actually used within the datasource as well, looks like it isn't now.

**Review by marefr — APPROVED** [review]
LGTM with nits. Doesn't really change anything so fine to merge I guess if you want

File: pkg/tsdb/grafana-testdata-datasource/kinds/query_test.go (hunk 3 above)
Comment (ryantxu) [reply pkg/tsdb/grafana-testdata-datasource/kinds/query_test.go:61]: > Nit. We should use proper go codegen rather thsn tests.

Yea -- lets explore a better path for that in a different PR.  This approach works and is easy to keep in sync -- but an explicit command is likely more idiomatic

**Comment by ryantxu** [comment]
@marefr -- I updated this so that testdata is now using the structs.  This is a much bigger change, so requesting another review 😜  -- it seems the cue definitions did not actually produce or reflect the values that were actually being used so I needed to make a bunch of changes

File: public/app/plugins/datasource/grafana-testdata-datasource/dataquery.gen.ts [hunk 4]
```diff
@@ -1,12 +1,4 @@
-// Code generated - EDITING IS FUTILE. DO NOT EDIT.
-//
-// Generated by:
-//     public/app/plugins/gen.go
-// Using jennies:
-//     TSTypesJenny
-//     PluginTsTypesJenny
-//
-// Run 'make gen-cue' from repository root to regenerate.
+// Code was originally generated cue
```
Comment (marefr) [inline public/app/plugins/datasource/grafana-testdata-datasource/dataquery.gen.ts:1]: From now on it's manual?

**Review by marefr — APPROVED** [review]
LGTM (with a question)

File: public/app/plugins/datasource/grafana-testdata-datasource/dataquery.gen.ts (hunk 4 above)
Comment (ryantxu) [reply public/app/plugins/datasource/grafana-testdata-datasource/dataquery.gen.ts:1]: yup -- I updated the file names to make that more clear.

We have a good path for jsonschema >> typescript generation; but not wired up yet

**Pull request merged**